### DISCUSSION TITLE
Custom User Menu intead of showing Avatar

### DISCUSSION THREAD

**Discussion by luciano-professor** [filamentphp/filament#10444] [Help] [answered]
I would like to display some dynamic text (the name of the person) or custom view instead of showing the Avatar as the trigger to the dropdown user menu.

Something like that
![image](https://github.com/filamentphp/filament/assets/43418124/606cc434-fbca-426d-8c56-fbef9de8a88c)

**Comment by leandrocfe** [comment, marked as answer]
Hooks as `panels::user-menu.after` `panels::user-menu.before` can help you to customize the user menu.
https://filamentphp.com/docs/3.x/support/render-hooks#registering-render-hooks

Another way would be creating this file in your project `resources/views/vendor/filament-panels/components/user-menu.blade.php`
Copy/paste the original content and edit what you want [(Filament doesn't recommend it)](https://filamentphp.com/docs/3.x/support/style-customization#publishing-blade-views).

**Reply by luciano-professor** [reply]
Thanks.

Using only the hooks I was not able to do what I want. 

But using the second alternative (that I know is not the ideal) I could do that. 

Maybe in a future, somebody PR something that enable a full customization on the user menu, allowing to not using the avatar and instead showing a custom component in its place. 

I cannot do this PR because I'm not well versed yet in Filament.

**Reply by cawecoy** [reply]
Oi professor (:

I had the same problem, but I've overrided `resources/views/vendor/filament-panels/components/avatar/user.blade.php` with the content bellow in my project. As it's a simpler file, it looks safer to override.

```
{{ filament()->auth()->user()->name }}
```

You can add a custom icon like:

```
{{ filament()->auth()->user()->name }}
@svg('icon-custom')
```

**Reply by henryonsoftware** [reply]
@cawecoy You saved my day!
